docs/SemanticARC.md

```diff
@@ -128,8 +145,41 @@ In order to simplify this, we will make the following changes:
    * @unowned @safe
    * @forwarding
 
+[EE] We really need a good definition of all these conventions here.
+I strongly believe that the local/global notation would help a lot for a good formal definition of the conventions.
+It's hard for the compiler to reason anything about global RCs. But local RCs are easily statically computable and for me, semantic ARC is all about dealing with local RCs.
+
+So we could define:
+
+@owned operand: local RC must be 1 and is decremented to 0 by the instruction, potentially destroying the object.
+@owned result: Returns the object with a local RC of 1
+@guaranteed operand: local RC must be 1 and is not modified by the instruction
+@unowned operand: local RC must be 0 and is not modified by the instruction.
+@unowned result: Returns the object with a local RC of 0
+@forwarding: Does not change the local RC which can be either 0 or 1
+
+Note that the local RC can only be 0 or 1. It doesn't make sense to have a local RC > 1.
```

Comment:
My main issue with this is that this is information that the optimizer can recover easily from the nature of a pairing. Why should we represent redundant information here.